pkg/common/workbuilder/workbuilder.go

```diff
@@ -0,0 +1,318 @@
+package workbuilder
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"open-cluster-management.io/addon-framework/pkg/common/workapplier"
+	workapiv1 "open-cluster-management.io/api/work/v1"
+)
+
+// DefaultManifestLimit is the max size of manifests data which is 50k bytes by default.
+const DefaultManifestLimit = 50 * 1024
+
+// DefaultManifestThreshold is the threshold size of manifests for the first new created manifestWork.
+const DefaultManifestThreshold = 0.8
+
+type manifestKey struct {
+	gvk             schema.GroupVersionKind
+	namespace, name string
+}
+
+type requiredManifestMapper map[manifestKey]workapiv1.Manifest
+
+// GetExistingManifestWorks is to get existing manifestWorks which need to be updated.
+type GetExistingManifestWorks func() ([]workapiv1.ManifestWork, error)
+
+// GenerateManifestWork is to generate a new manifestWork with empty manifest, need to make sure the name is unique.
+type GenerateManifestWork func(index int) workapiv1.ManifestWork
```

Comment:
GenerateManifestWorkObjectMeta and only return metav1.ObjectMeta

Reply:
done